doc/modules/ensemble.rst

```diff
@@ -1,534 +1,437 @@
+then samples with missing values are mapped to whichever child has the most
+samples.
 
--  Efficient binning: HGBT uses an efficient binning algorithm that can handle
-   large datasets with a high number of features. The binning algorithm can
-   pre-process the data to speed up the subsequent tree construction (see
-   :ref:`Why it's faster <Why_it's_faster>`). In contrast, the scikit-learn
-   implementation of random forests does not use binning and relies on exact
-   splitting, which can be computationally expensive.
+.. _sw_hgbdt:
 
-Overall, the computational cost of HGBT versus RF depends on the specific
-characteristics of the dataset and the modeling task. It's always a good idea
-to try both models and compare their performance and computational efficiency
-on your specific problem to determine which model is the best fit.
+Sample weight support
+^^^^^^^^^^^^^^^^^^^^^
 
-.. topic:: Examples:
+:class:`HistGradientBoostingClassifier` and
+:class:`HistGradientBoostingRegressor` sample support weights during
+:term:`fit`.
 
- * :ref:`sphx_glr_auto_examples_ensemble_plot_forest_hist_grad_boosting_comparison.py`
+The following toy example demonstrates how the model ignores the samples with
+zero sample weights:
 
-Extremely Randomized Trees
---------------------------
+    >>> X = [[1, 0],
+    ...      [1, 0],
+    ...      [1, 0],
+    ...      [0, 1]]
+    >>> y = [0, 0, 1, 0]
+    >>> # ignore the first 2 training samples by setting their weight to 0
+    >>> sample_weight = [0, 0, 1, 1]
+    >>> gb = HistGradientBoostingClassifier(min_samples_leaf=1)
+    >>> gb.fit(X, y, sample_weight=sample_weight)
+    HistGradientBoostingClassifier(...)
+    >>> gb.predict([[1, 0]])
+    array([1])
+    >>> gb.predict_proba([[1, 0]])[0, 1]
+    0.99...
 
-In extremely randomized trees (see :class:`ExtraTreesClassifier`
-and :class:`ExtraTreesRegressor` classes), randomness goes one step
-further in the way splits are computed. As in random forests, a random
-subset of candidate features is used, but instead of looking for the
-most discriminative thresholds, thresholds are drawn at random for each
-candidate feature and the best of these randomly-generated thresholds is
-picked as the splitting rule. This usually allows to reduce the variance
-of the model a bit more, at the expense of a slightly greater increase
-in bias::
+As you can see, the `[1, 0]` is comfortably classified as `1` since the first
+two samples are ignored due to their sample weights.
 
-    >>> from sklearn.model_selection import cross_val_score
-    >>> from sklearn.datasets import make_blobs
-    >>> from sklearn.ensemble import RandomForestClassifier
-    >>> from sklearn.ensemble import ExtraTreesClassifier
-    >>> from sklearn.tree import DecisionTreeClassifier
+Implementation detail: taking sample weights into account amounts to
+multiplying the gradients (and the hessians) by the sample weights. Note that
```

Comment:
```suggestion
multiplying the gradients (and the hessians) by the sample weights.
```
It feels like "Implementation detail" already puts us in "Note something ..." mode, so having another "Note" inside a note is like a segue inside a segue